### PR TITLE
Rule validation tests

### PR DESCRIPTION
**Pull request by davidwengier**
Thanks to Drew for the suggestion, these tests validate the rules for rules, and will ensure we don't regress things. Plan is to add more as more things are fixed up.

Includes #4635 so ignore the first 4 commits, and/or just ignore all of the `.xaml` files if its easier since this also includes some fixing of things found by these tests.

Part of #4105

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/XamlRuleTests.cs [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+
+        [Theory]
+        [MemberData(nameof(GetAllRules))]
+        public void NonVisiblePropertiesShouldntBeLocalized(string ruleName, string fullPath)
+        {
+            XmlDocument rule = LoadXamlRule(fullPath);
+
+            foreach (XmlNode node in rule.DocumentElement.ChildNodes)
+            {
+                if (node is XmlElement element && element.HasAttribute("Visible") && element.Attributes["Visible"].Value.Equals("false", StringComparison.OrdinalIgnoreCase))
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/XamlRuleTests.cs]: Is a property that omits the `Visible` attribute considered visible?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/XamlRuleTests.cs [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+            // First fix up the Name and DisplayName as we know they'll differ.
+            rule.DocumentElement.Attributes["Name"].Value = "None";
+            rule.DocumentElement.Attributes["DisplayName"].Value = "General";
+
+            AssertXmlEqual(none, rule);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetAllRules))]
+        public void NonVisiblePropertiesShouldntBeLocalized(string ruleName, string fullPath)
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/XamlRuleTests.cs]: Do we need another test that verifies visible properties have `DisplayName` (and possibly other fields)?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/XamlRuleTests.cs (hunk 1 above)
Comment (davidwengier) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/XamlRuleTests.cs]: Yes, the default is true.

https://github.com/Microsoft/msbuild/blob/e70a3159d64f9ed6ec3b60253ef863fa883a99b1/src/Framework/XamlTypes/BaseProperty.cs#L52

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/XamlRuleTests.cs (hunk 2 above)
Comment (davidwengier) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/XamlRuleTests.cs]: Yes, and many more to boot

Comment (drewnoakes) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/XamlRuleTests.cs]: Would be good to capture your ideas for such tests in a tracking issue, if they're not already.

Comment (davidwengier) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/XamlRuleTests.cs]: I'm generally tracking everything in #4105

**Pull request merged**